### PR TITLE
fix!: avoid panicking on CancelUnbondingDelegation 

### PR DESCRIPTION
**Pull request by mpoke**
## Description

Closes: https://github.com/cosmos/interchain-security/issues/948

Avoids panicking the provider when an unbonding delegation was removed through a `CancelUnbondingDelegation` message. 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [x] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [x] Added `!` to the type prefix if API or client breaking change
* [x] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))
* [x] Provided a link to the relevant issue or specification
* [x] Followed the guidelines for [building SDK modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules)
* [ ] Included the necessary unit and integration [tests](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#testing)
  * [ ] TODO: add a e2e test for checking this behavior
* [ ] Added a changelog entry to `CHANGELOG.md`
* [x] Included comments for [documenting Go code](https://blog.golang.org/godoc)
* [ ] Updated the relevant documentation or specification
* [x] Reviewed "Files changed" and left comments if necessary 
* [ ] Confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

* [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] confirmed `!` in the type prefix if API or client breaking change
* [ ] confirmed all author checklist items have been addressed 
* [ ] reviewed state machine logic
* [ ] reviewed API design and naming
* [ ] reviewed documentation is accurate
* [ ] reviewed tests and test coverage

### CODE REVIEW

File: testutil/keeper/mocks.go [hunk 1]
```diff
@@ -98,6 +98,14 @@ func (m *MockStakingKeeper) GetLastValidators(ctx types0.Context) []types5.Valid
 	return ret0
 }
 
+func (m *MockStakingKeeper) GetUnbondingType(ctx types0.Context, id uint64) (types5.UnbondingType, bool) {
```
Comment (smarshall-spitzbart) [inline testutil/keeper/mocks.go:101]: Is this needed for the PR?

File: x/ccv/types/expected_keepers.go [hunk 2]
```diff
@@ -46,6 +46,7 @@ type StakingKeeper interface {
 	MaxValidators(ctx sdk.Context) uint32
 	GetLastTotalPower(ctx sdk.Context) math.Int
 	GetLastValidators(ctx sdk.Context) (validators []stakingtypes.Validator)
+	GetUnbondingType(ctx sdk.Context, id uint64) (unbondingType stakingtypes.UnbondingType, found bool)
```
Comment (smarshall-spitzbart) [inline x/ccv/types/expected_keepers.go:49]: Same, is this needed?

File: x/ccv/provider/keeper/relay.go [hunk 3]
```diff
@@ -103,8 +103,18 @@ func (k Keeper) completeMaturedUnbondingOps(ctx sdk.Context) {
 		// Attempt to complete unbonding in staking module
 		err := k.stakingKeeper.UnbondingCanComplete(ctx, id)
 		if err != nil {
-			// UnbondingCanComplete fails if the unbonding operation was not found,
-			// which means that the state of the x/staking module of cosmos-sdk is invalid.
+			if stakingtypes.ErrUnbondingNotFound.Is(err) {
+				// The unbonding was not found.
```
Comment (smarshall-spitzbart) [inline x/ccv/provider/keeper/relay.go:107]: It's a bit unclear to me why `k.stakingKeeper.UnbondingCanComplete` would return `stakingtypes.ErrUnbondingNotFound` while `k.stakingKeeper.GetUnbondingType(ctx, id)` would return found as true?

File: x/ccv/types/expected_keepers.go (hunk 2 above)
Comment (mpoke) [reply x/ccv/types/expected_keepers.go:49]: Yes, see `completeMaturedUnbondingOps`.

File: x/ccv/provider/keeper/relay.go (hunk 3 above)
Comment (mpoke) [reply x/ccv/provider/keeper/relay.go:107]: Have a look at https://github.com/cosmos/cosmos-sdk/blob/v0.45.16-ics/x/staking/keeper/unbonding.go#L218

**Pull request merged**